src/server/editorServices.ts

```diff
@@ -0,0 +1,1646 @@
+                        files: parsedCommandLine.fileNames,
+                        compilerOptions: parsedCommandLine.options
+                    };
+                    if (rawConfig.formatCodeOptions) {
+                        projectOptions.formatCodeOptions = rawConfig.formatCodeOptions;
+                    }
+                    proj.setProjectOptions(projectOptions);
+                    return { success: true, project: proj };
+                }
+                else {
+                    return { errorMsg: "no files found" };
+                }
+            }
+        }
+
+        createProject(projectFilename: string) {
+            var eproj = new Project(this);
+            eproj.projectFilename = projectFilename;
+            return eproj;
+        }
+
+    }
+
+    class CompilerService {
+        host: LSHost;
+        languageService: ts.LanguageService;
+        classifier: ts.Classifier;
+        settings = ts.getDefaultCompilerOptions();
+        documentRegistry = ts.createDocumentRegistry();
+        formatCodeOptions: ts.FormatCodeOptions = CompilerService.defaultFormatCodeOptions;
+
+        constructor(public project: Project) {
+            this.host = new LSHost(project.projectService.host, project);
+            // override default ES6 (remove when compiler default back at ES5)
+            this.settings.target = ts.ScriptTarget.ES5;
+            this.host.setCompilationSettings(this.settings);
+            this.languageService = ts.createLanguageService(this.host, this.documentRegistry);
+            this.classifier = ts.createClassifier();
+        }
+
+        setCompilerOptions(opt: ts.CompilerOptions) {
+            this.settings = opt;
+            this.host.setCompilationSettings(opt);
+        }
+
+        isExternalModule(filename: string): boolean {
+            var sourceFile = this.languageService.getSourceFile(filename);
+            return ts.isExternalModule(sourceFile);
+        }
+
+        static defaultFormatCodeOptions: ts.FormatCodeOptions = {
+            IndentSize: 4,
+            TabSize: 4,
+            NewLineCharacter: ts.sys.newLine,
+            ConvertTabsToSpaces: true,
+            InsertSpaceAfterCommaDelimiter: true,
+            InsertSpaceAfterSemicolonInForStatements: true,
+            InsertSpaceBeforeAndAfterBinaryOperators: true,
+            InsertSpaceAfterKeywordsInControlFlowStatements: true,
+            InsertSpaceAfterFunctionKeywordForAnonymousFunctions: false,
+            InsertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis: false,
+            PlaceOpenBraceOnNewLineForFunctions: false,
+            PlaceOpenBraceOnNewLineForControlBlocks: false,
+        }
+
+    }
+
+    interface LineCollection {
+        charCount(): number;
+        lineCount(): number;
+        isLeaf(): boolean;
+        walk(rangeStart: number, rangeLength: number, walkFns: ILineIndexWalker): void;
+    }
+
+    export interface ILineInfo {

```

Comment:
Can we put this all into its own text API or something?  I feel like this is all private implementation details that should not be exported or used by anything else.